### PR TITLE
Add a workflow to control which apps are released

### PR DESCRIPTION
**Pull request by ian-noaa**
Remove the ability for the "Build" workflow to push to the
production image registry. Create a new workflow that copies
tagged images from the dev registry to the production registry
when a GitHub Release is created.

### CODE REVIEW

**Comment by ian-noaa** [comment]
Hey @mollybsmith-noaa  & @randytpierce - I wanted you to take a peek at this. This changes our CI so that images are only pushed to the production registry if they're listed in the `deploy.yaml` file. It also changes CI so that the images are only pushed to the production registry when a [GitHub release](https://docs.github.com/en/repositories/releasing-projects-on-github/managing-releases-in-a-repository) is created from an existing tag. 

This means our release process would involve creating the appropriate tag on `main` and then going to the GitHub UI and creating the release announcement there. I think this could be useful since GitHub releases can do things like auto generate release notes that you can then edit and review before publishing the release.

**Comment by mollybsmith-noaa** [comment]
This code all looks fine but I would like to have a meeting this week to fully discuss how the whole Github build workflow will operate, and how we'll do the changeover. As the person who will be doing all the app builds and releases I need to understand how all of this works and how to take care of it. I'm going to hold off on officially approving the issue until after that meeting, but again, the code itself looks good.

**Comment by ian-noaa** [comment]
@mollybsmith-noaa & @randytpierce - given our discussion about the GitHub workflow yesterday am I okay to merge this?

**Review by mollybsmith-noaa — APPROVED** [review]
Yes, go ahead. Are there similar pull requests that need approval on MATScommon or METexpress?

**Comment by ian-noaa** [comment]
Thanks! There aren't yet. However, I will need to add one to METexpress to keep it in sync.

**Pull request merged**
